### PR TITLE
fix: migrate package options to new format

### PR DESCRIPTION
**Pull request by anshumanv**
**What kind of change does this PR introduce?**
refactor

**Did you add tests for your changes?**
Yes

**If relevant, did you update the documentation?**
NA

**Summary**
Migrates the command options to new format

**Does this PR introduce a breaking change?**
No

**Other information**

### CODE REVIEW

**Comment by anshumanv** [comment]
wip on core

File: packages/webpack-cli/lib/webpack-cli.js [hunk 1]
```diff
@@ -382,92 +398,153 @@ class WebpackCLI {
...
+                ],
                 alias: 'j',
                 description: 'Prints result as JSON or store it in a file.',
             },
 
             // For webpack@4
             {
                 name: 'entry',
-                type: String,
+
```
Comment (alexander-akait) [inline packages/webpack-cli/lib/webpack-cli.js:460]: Missing `type` and `configs`

Comment (anshumanv) [reply packages/webpack-cli/lib/webpack-cli.js]: added - https://github.com/webpack/webpack-cli/pull/2508/commits/623c17a5c353b59f524d327a30968cb2b5843de5

**Comment by alexander-akait** [comment]
@anshumanv Can we rebase and fix CI?

**Comment by anshumanv** [comment]
wip on fix soon

**Comment by anshumanv** [comment]
need to handle defaults for generator

**Comment by anshumanv** [comment]
@alexander-akait CI green ⭐

**Review by alexander-akait — APPROVED** [review]
/cc @webpack/cli-team

**Review by snitin315 — APPROVED** [review]

**Pull request merged**